### PR TITLE
Optimize app entrypoint

### PR DESCRIPTION
**Pull request by bramkragten**
## Proposed change

Optimize the size of the app entrypoint, now down to ~335kb (when Polymer is removed)

image

## Type of change

- [ ] Dependency upgrade
- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (thank you!)
- [ ] Breaking change (fix/feature causing existing functionality to break)
- [X] Code quality improvements to existing code or addition of tests

## Example configuration

```yaml

```

## Additional information

- This PR fixes or closes issue: fixes #
- This PR is related to issue or discussion:
- Link to documentation pull request:

## Checklist

- [ ] The code change is tested and works locally.
- [ ] There is no commented out code in this PR.
- [ ] Tests have been added to verify that the new code works.

If user exposed functionality or configuration variables are added/changed:

- [ ] Documentation added/updated for [www.home-assistant.io][docs-repository]

[docs-repository]: https://github.com/home-assistant/home-assistant.io

### CODE REVIEW

File: src/layouts/hass-router-page.ts [hunk 1]
```diff
@@ -256,10 +254,12 @@ export class HassRouterPage extends ReactiveElement {
   }
 
   protected createLoadingScreen() {
+    import("./hass-loading-screen");
```
Comment (steverep) [inline src/layouts/hass-router-page.ts:257]: Does this not have to be awaited?  Does lit pick up properties being set downstream (before the element is defined)?

File: src/layouts/hass-router-page.ts [hunk 2]
```diff
@@ -256,10 +254,12 @@ export class HassRouterPage extends ReactiveElement {
...
     return document.createElement("hass-loading-screen");
   }
 
   protected createErrorScreen(error: string) {
+    import("./hass-error-screen");
```
Comment (steverep) [inline src/layouts/hass-router-page.ts:262]: Same here

File: src/layouts/ha-init-page.ts [hunk 3]
```diff
@@ -54,10 +49,15 @@ class HaInitPage extends LitElement {
     }
   }
 
+  protected updated(changedProperties: PropertyValues) {
```
Comment (steverep) [inline src/layouts/ha-init-page.ts:52]: Should start the import before rendering, no?

```suggestion
  protected willUpdate(changedProperties: PropertyValues<this>) {
```

File: src/layouts/hass-router-page.ts (hunk 1 above)
Comment (bramkragten) [reply src/layouts/hass-router-page.ts:257]: Yes, it will be upgraded ones defined

**Review by balloob — DISMISSED** [review]

**Review by balloob — APPROVED** [review]

**Review by steverep — APPROVED** [review]

**Pull request merged**